lib/persisted-model.js

```diff
@@ -573,9 +605,41 @@ module.exports = function(registry) {
       accepts: { arg: 'data', type: 'object', http: { source: 'body' }, description:
         'Model instance data' },
       returns: { arg: 'data', type: typeName, root: true },
-      http: { verb: 'put', path: '/' },
+      http: { verb: 'patch', path: '/' },
     });
 
+    if (!options.replaceOnPUT) {
+      setRemoting(PersistedModel, 'upsert', {
```

Comment:
I don't think it's correct to call `setRemoting` twice for the same method, expecting it to create two HTTP endpoints. The correct way is to provide an array of values in the `http` property. See how `exists` defines two HTTP endpoints:

```js
       http: [
         { verb: 'get', path: '/:id/exists' },
         { verb: 'head', path: '/:id' },
       ],
```

We should do the same here. For example:

```js
var upsertEndpoints = [{ verb: 'patch', path: '/' }];
if (!options.replaceOnPUT) {
  upsertEndpoints.push( {
    verb: 'put',
    patch: '/',
  });
}

setRemoting('PersistedModel', 'upsert', {
  // ...
  http: upsertEndpoints
});
```

Or you can use the approach you have in place for `replaceOrCreateOptions` below, both ways are fine.